### ISSUE TITLE
Memory leak: Navigating in the TOC (DocumentationReader) gets slower after subsequent clicks

### ISSUE THREAD

**Issue by michael** [substance/substance#336]
Investigate.

**Comment by michael** [comment]
Found out what is the root of the evil:

https://github.com/substance/substance/blob/master/ui/TOCPanel.js#L53

It seems that because the owner is confused and I manually need bind the click handler to `this` the handler gets attached on each rerender, causing the slowdown we observe. I can workaround this via jQuery click handler but we should get this fixed soon via #313.

**Comment by michael** [comment]
Is it possible that component click handlers are leaking always?

``` js
this.render = function() {
  var that = this;

  return $$('a').append('MEEH')
    .attr('href', '#')
    .on('click', function() {
      console.log('clicked');
      that.rerender();
    });
}
```

This causes a lot of 'clicked' logs.

**Comment by obuchtala** [comment]
With the current issue-312 branch I have fixed the owner/parent issues.
It seems to me that this is resolved with that.
@michael could you try out plz?

**Comment by obuchtala** [comment]
Fixed via #346
